client/state/mailchimp/lists/reducer.js

```diff
@@ -0,0 +1,22 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+
+import { combineReducers, createReducer } from 'state/utils';
+import { MAILCHIMP_LISTS_RECEIVE } from 'state/action-types';
+
+export const items = createReducer(
+	{},
+	{
+		[ MAILCHIMP_LISTS_RECEIVE ]: ( state, { siteId, lists } ) => ( {
+			...state,
+			[ siteId ]: lists,
```

Comment:
Requesting and storing something for each site separately is very common in Calypso. Did you see the `keyedReducer` helper that can make your reducer smaller and less repetitive?